File: lib/models/Comment.dart
import 'User.dart';
import 'Story.dart';

class Comment {
  int? id;
  User? user;
  Story? story;
  String? content;
  DateTime? createdAt;
  DateTime? update_at;
  int? parentId;
   List<Comment>? replies =[];

  Comment({
    this.id,
    this.user,
    this.story,
    this.content ,
    this.createdAt,
    this.update_at,
    this.parentId,

  });



  Comment.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    user = json['user'] != null ? User.fromJson(json['user']) : null;
    story = json['story'] != null ? Story.fromJson(json['story']) : null;
    content = json['content'];
    createdAt = json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null;
    update_at = json['updateAt'] != null ? DateTime.parse(json['updateAt']) : null;
    parentId = json['parentId'];
    // Xóa phần parse replies ở đây
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    if (user != null) {
      data['user'] = user!.toJson();
    }
    if (story != null) {
      data['story'] = story!.toJson();
    }
    data['content'] = content;
    data['createdAt'] = createdAt?.toIso8601String();
    data['updateAt'] = update_at?.toIso8601String();
    data['parentId'] = parentId;
    // Xóa phần replies ở đây
    return data;
  }
}

File: src/main/java/com/example/noveltruyen/Model/Comment.java
package com.example.noveltruyen.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "comments")
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    // Thay đổi kiểu dữ liệu thành Long và thêm mapping
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    // Thay đổi kiểu dữ liệu thành Long và thêm mapping
    @ManyToOne
    @JoinColumn(name = "story_id", nullable = false)
    private Story story;

    @Lob
    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @CreationTimestamp
    private LocalDateTime updateAt;

    // Thêm các trường để hỗ trợ comment phản hồi

    @ManyToOne
    @JoinColumn(name = "parent_id")
    private Comment parent; // Tham chiếu đến comment cha (nếu là comment phản hồi)

    @OneToMany(mappedBy = "parent", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> replies; // Danh sách các comment phản hồi
}